src/GenLevelStudiesHists.cc

```diff
@@ -115,6 +144,10 @@ void GenLevelStudiesHists::fill(const RecoEvent & event){
       hist<TH1F>("4Lpt")->Fill(FourLeptons.Pt(), weight);
     }
 
+    if (m.status()!=1 || !m.get_statusflag(GenParticle::isLastCopy)) continue;
+    hist<TH1F>("PdgId")->Fill(m.pdgid(), weight);
+    hist<TH1F>("PdgIdZoom")->Fill(m.pdgid(), weight);
```

Comment:
you should really check the PR that I did. having one bin with label per pdgID is much easier to read instead of 10k bins